src/api/crypto/blockchain/BalanceOracle.cpp

```diff
@@ -214,50 +214,35 @@ auto BalanceOracle::Imp::process_registration(Message&& in) noexcept -> void
     const auto haveNym = (2 < body.size());
     auto output = opentxs::blockchain::Balance{};
     const auto& chainFrame = body.at(1);
-    const auto nym = [&] {
-        if (haveNym) {
-
-            return api_.Factory().NymID(body.at(2));
-        } else {
+    const auto nym =
+        haveNym ? api_.Factory().NymID(body.at(2)) : api_.Factory().NymID();
 
-            return api_.Factory().NymID();
-        }
-    }();
     const auto chain = chainFrame.as<Chain>();
+
     const auto postcondition = ScopeGuard{[&]() {
-        router_.Send([&] {
-            auto work = opentxs::network::zeromq::tagged_reply_to_message(
-                in, WorkType::BlockchainBalance);
-            work.AddFrame(chainFrame);
-            output.first.Serialize(work.AppendBytes());
-            output.second.Serialize(work.AppendBytes());
+        auto work = opentxs::network::zeromq::tagged_reply_to_message(
+            in, WorkType::BlockchainBalance);
+        work.AddFrame(chainFrame);
+        output.first.Serialize(work.AppendBytes());
+        output.second.Serialize(work.AppendBytes());
 
-            if (haveNym) { work.AddFrame(nym); }
+        if (haveNym) { work.AddFrame(nym); }
 
-            return work;
-        }());
+        router_.Send(std::move(work));
     }};
+
     const auto unsupported =
         (0 == opentxs::blockchain::SupportedChains().count(chain)) &&
         (Chain::UnitTest != chain);
 
     if (unsupported) { return; }
 
-    auto& subscribers = [&]() -> auto&
-    {
-        auto& chainData = data_[chain];
+    auto& chainData = data_[chain];
+    const auto& subscribers =
```

Comment:
Why did you use a const reference to a pointer here? It seems unnecessary and potentially misleading.